From 2d31a8500edb755dbf253656f41a34c55223438e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20Br=C3=BCns?= <stefan.bruens@rwth-aachen.de>
Date: Sun, 7 Oct 2018 00:36:07 +0200
Subject: [PositionCodec] Avoid crash in case of corrupt data

Summary:
getDifferentialVarInt* returns a nullptr in case of a decoding error.

See also D11825.

BUG: 367480

Reviewers: #frameworks, poboiko, ngraham, #baloo

Reviewed By: ngraham, #baloo

Subscribers: kde-frameworks-devel, #baloo

Tags: #frameworks, #baloo

Differential Revision: https://phabricator.kde.org/D15993
---
 src/codecs/positioncodec.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/codecs/positioncodec.cpp b/src/codecs/positioncodec.cpp
index 572242e..d0eaf12 100644
--- a/src/codecs/positioncodec.cpp
+++ b/src/codecs/positioncodec.cpp
@@ -54,6 +54,9 @@ QVector<PositionInfo> PositionCodec::decode(const QByteArray& arr)
         info.docId = decodeFixed64(data);
         data += sizeof(quint64);
         data = getDifferentialVarInt32(data, end, &info.positions);
+        if (!data) {
+            return QVector<PositionInfo>();
+        }

         vec << info;
     }
--
cgit v0.11.2
